views/errors.go

```diff
@@ -1,14 +1,35 @@
 package views
 
 import (
-	"fmt"
-	"github.com/ystv/web-auth/public/templates"
-	"net/http"
+	"errors"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/gommon/log"
+	"github.com/ystv/web-auth/templates"
 )
 
-func (v *Views) Error404(w http.ResponseWriter, _ *http.Request) {
-	err := v.template.RenderNoNavsTemplate(w, nil, templates.NotFound404Template)
-	if err != nil {
-		fmt.Println(err)
+func (v *Views) CustomHTTPErrorHandler(err error, c echo.Context) {
+	log.Print(err)
+	var he *echo.HTTPError
+	var status int
+	if errors.As(err, &he) {
+		status = he.Code
+	} else {
+		status = 500
 	}
+	c.Response().WriteHeader(status)
+	data := struct {
+		Code  int
+		Error interface{}
+	}{
+		Code:  status,
+		Error: he.Message,
+	}
+	err1 := v.template.RenderTemplate(c.Response().Writer, data, templates.ErrorTemplate, templates.NoNavType)
```

Comment:
Do we actually set the status code in the HTTP response somewhere, or will the caller get a 200?

Reply:
This line adds it to the response `c.Response().WriteHeader(status)` in that function, it took a while to figure out but does persist